contrib/completion/bash/docker

```diff
@@ -5503,6 +5500,18 @@ _docker_compose() {
 	COMPREPLY=( $(compgen -W "${result}" -- "$current") )
 }
 
+_docker_compose() {
```

Comment:
This should be generic.
Assuming that a plugin contributes only one top level command, and that this command equals its name, we can check whether the command is part of `$PLUGINS_INSTALLED` right before calling https://github.com/docker/cli/blob/304a2dca5fd9d203ed618fc73066612eb2ce0e7a/contrib/completion/bash/docker#L5663 and if so, call a generic plugin completion instead.

Reply:
Ah, thanks! Yes, I was assuming something like that was possible. That said, plugins do provide subcommands, in which case it should call (e.g. for `docker buildx imagetools`)

(looks like a `""` is needed to force it to look up subcommands);

```bash
docker __complete buildx imagetools ""
create	Create a new image based on source images
inspect	Show details of image in the registry
:4
Completion ended with directive: ShellCompDirectiveNoFileComp
```

Or, for flags (`--`);

```bash
docker __complete buildx imagetools --
--builder	Override the configured builder instance
:4
Completion ended with directive: ShellCompDirectiveNoFileComp

Reply:
I'm a big fan of a generic plugin-completion solution.  That being said, I think care is needed here.

The `$PLUGINS_INSTALLED` list returns all cli plugins, not just the ones provided by docker.  I don't think it would be right to call `docker __completeNoDesc <anyPlugin>` as those plugins may not support the `__completeNoDesc` command and may instead simply ignore it and execute whatever functionality they normally perform.

This was already hinted at by @thaJeztah earlier with the idea of "a `v0.0.2` version of the metadata that indicates that the plugin supports the __complete subcommands" (https://github.com/docker/cli/pull/3254#discussion_r689396570), so sorry if I'm repeating things; I thought it was worth mentioning as it has been an important point in the discussion about how to support plugin completion for `kubectl`.